flink-runtime/src/test/java/org/apache/flink/runtime/blob/BlobCacheSizeTrackerTest.java

```diff
@@ -0,0 +1,286 @@
+
+        final BlobKey.BlobType blobType = BlobKey.BlobType.PERMANENT_BLOB;
+
+        final Configuration config = new Configuration();
+        config.setString(
+                BlobServerOptions.STORAGE_DIRECTORY, temporaryFolder.newFolder().getAbsolutePath());
+
+        // The size limit is the size of 2 blobs
+        int sizeLimit = 2;
+
+        try (BlobServer server = new BlobServer(config, new VoidBlobStore());
+                BlobCacheService cache =
+                        initBlobCacheServiceWithSizeLimit(
+                                config,
+                                new VoidBlobStore(),
+                                new InetSocketAddress("localhost", server.getPort()),
+                                sizeLimit * blobSize)) {
+
+            server.start();
+
+            // Initialize the data structures
+            JobID[] jobId = new JobID[4];
+            byte[][] data = new byte[4][blobSize];
+            BlobKey[] key = new BlobKey[4];
+            File[] file = new File[4];
+
+            // Prepare the blob data
+            byte[] blobData = new byte[blobSize];
+            random.nextBytes(blobData);
+
+            for (int i = 0; i < 4; i++) {
+                jobId[i] = new JobID();
+                data[i] = Arrays.copyOf(blobData, blobData.length);
+                data[i][i] ^= 1;
+            }
+
+            // Put the first, the second and the the third BLOB into the blob server one by one
+            // and let the cache retrieve the data from the blob
+            for (int i = 0; i < 3; i++) {
+                key[i] = put(server, jobId[i], data[i], blobType);
+                assertNotNull(key[i]);
+
+                readFileAndVerifyContent(cache, jobId[i], key[i], data[i]);
+                file[i] = getFile(cache, jobId[i], key[i]);
+                assertTrue(file[i].exists());
+            }
+
+            // Since the size limit of the cache is the size of 2 blobs,
+            // the first blob is removed and the second one remains
+            assertFalse(file[0].exists());
+            assertTrue(file[1].exists());
+
+            // Get the second blob once again, make the third blob to be the last recently used
+            readFileAndVerifyContent(cache, jobId[1], key[1], data[1]);
+
+            // Then get the fourth blob, make sure the third blob is replaced.
+            // Here we cannot get the first blob again, because for transient blob,
+            // BlobServer will delete the blob once it's transmitted.
+            key[3] = put(server, jobId[3], data[3], blobType);
+            readFileAndVerifyContent(cache, jobId[3], key[3], data[3]);
+            file[3] = getFile(cache, jobId[3], key[3]);
+
+            assertTrue(file[1].exists());
+            assertTrue(file[3].exists());
+            assertFalse(file[2].exists());
+        }
+    }
+
+    @Test
+    public void testGetPermanentCacheWithSizeLimitConcurrently() throws Exception {
```

Comment:
It is test of `PermanentBlobCache` and should not be in `BlobCacheSizeTrackerTest`.